purchase_packaging/models/procurement_order.py

```diff
@@ -50,10 +50,18 @@ def subtract_procurements_from_orderpoints(self):
         # This override will return 12 and no additionnal procurement will be
         # created
         res = super(Orderpoint, self).subtract_procurements_from_orderpoints()
+        procurements = self.env['procurement.order'].search(
+            [('orderpoint_id', 'in', self.ids),
+             ('state', 'not in', ['cancel', 'done']),
+             ('purchase_line_id.state', '=', ['draft', 'sent', 'to approve'])])
+        procurements.mapped('product_uom.rounding')
+        procurements.mapped('purchase_line_id.state')
```

Comment:
Why this two `mapped`??

Reply:
@lreficent That's done to be sure all data being fetched in recordset before entering the loop (that's the case on functions that don't depends on ...'depends' - which makes pre-fetching).

That's a good way to avoid too much queries.

Reply:
It prefetch data and put them in memory. Then the following code run must faster.

Reply:
@tfossoulw @rousseldenis Thanks for the explanation, good to know it!